plugin/core/sessions.py

```diff
@@ -850,6 +851,9 @@ def m_workspace_configuration(self, params: Dict[str, Any], request_id: Any) ->
         for requested_item in requested_items:
             configuration = self.config.settings.copy(requested_item.get('section') or None)
             if self._plugin:
+                buffer_view = self.get_session_buffer_for_uri_async(requested_item['scopeUri'])
+                if buffer_view:
+                    requested_item['languageId'] = buffer_view.language_id
```

Comment:
Does this have the possibility to ever clash? What if there exists a server that asks for the `languageId` section?

I think clashes can be avoided by putting it in under an `_LSP` key and stashing all of the extra information in that dict... something like `requested_item['_LSP'] = {'languageId': buffer_view.language_id}`.

Reply:
> Does this have the possibility to ever clash? What if there exists a server that asks for the `languageId` section?

The `requested_item` has this structure:
```ts
interface ConfigurationItem {
	/**
	 * The scope to get the configuration section for.
	 */
	scopeUri?: DocumentUri;

	/**
	 * The configuration section asked for.
	 */
	section?: string;
}
```
So I'm not sure how it could clash, regardless of what server asks for. It could only clash if LSP spec would add a `languageId` key to that interface.

Regardless, I could add it under `_LSP` object I guess. Has kinda "internal" vibe to it and it would seems a bit wrong to access that from the plugin.

Reply:
You're right it probably will not result in clashes with the protocol.

Reply:
Regardless I think this is the wrong approach.